### PR TITLE
Feature - input / output rulemaps for linked rules

### PR DESCRIPTION
**Pull request by brysonjbest**
- [x] Linked rules recursively compile all inputs and outputs into the top-level rule
- [x] Update to fix bug on handling of csv generation for comma escapes

### CODE REVIEW

File: src/api/ruleMapping/ruleMapping.service.ts [hunk 1]
```diff
@@ -42,16 +60,22 @@ export class RuleMappingService {
...
-    return { [fieldKey]: fields };
+
+    const results = await Promise.all(promises);
+    const fields = results.flat();
+
+    const uniqueFieldsMap = new Map<string, any>();
+
+    fields.forEach((field) => {
+      uniqueFieldsMap.set(field.property, field);
+    });
```
Comment (timwekkenbc) [inline src/api/ruleMapping/ruleMapping.service.ts:71]: I think you can just use map here: `const uniqueFieldsMap = new Map(fields.map(field => [field.property, field]));`

File: src/api/ruleMapping/ruleMapping.service.spec.ts [hunk 2]
```diff
@@ -84,15 +84,171 @@ describe('RuleMappingService', () => {
...
+    });
+
+    it('should handle expressionNode with complex expressions correctly', async () => {
+      const nodes: Node[] = [
+        {
+          type: 'expressionNode',
+          content: {
+            expressions: [
+              { key: 'expr1', value: 'field3' },
+              { key: 'expr2', value: 'complexExpr + 2' },
```
Comment (timwekkenbc) [inline src/api/ruleMapping/ruleMapping.service.spec.ts:150]: First, thanks for these tests as I was wondering what the difference between a simple vs complex expression looked like. Second, I think the first one here is still a simple one though, right?

**Review by timwekkenbc — APPROVED** [review]
I think it looks good. This stuff just continues to get more and more annoyingly convoluted though - that's not a comment about your implementation though, just about the difficulty of approaching the problem this way. Alternative approaches such as where the rule author defines the input schema look more and more appealing.

File: src/api/ruleMapping/ruleMapping.service.ts (hunk 1 above)
Comment (brysonjbest) [reply src/api/ruleMapping/ruleMapping.service.ts:71]: Updated! Definitely cleaner, thanks :)

File: src/api/ruleMapping/ruleMapping.service.spec.ts (hunk 2 above)
Comment (brysonjbest) [reply src/api/ruleMapping/ruleMapping.service.spec.ts]: Updated so they are both complex. Initially had only one as then you could compare in the same test and see that it worked correctly while not impacting non-complex expressions, but it's sort of redundant with the simple expressions test so revised per your comment.

**Pull request merged**